rfc/rfc-81/rfc-81.md

```diff
@@ -0,0 +1,108 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-81: Log Compaction with Merge Sort
+
+## Proposers
+- @usberkeley
+
+## Approvers
+- @danny0405
+
+## Status
+JIRA: https://issues.apache.org/jira/browse/HUDI-8033
+
+## Abstract
+Add lightweight LogCompaction to improve the writing performance of the write side, and improve the query performance of the read side (Spark/Presto, etc.) in some scenarios without having to wait for heavy and time-consuming operations such as Compaction or Clustering.
+
+## Background
+The previous LogCompaction mainly merged log files through HoodieMergedLogRecordScanner, and used ExternalSpillableMap internally to achieve record merging, which resulted in performance loss of writing to disk, when the amount of log data exceeds merge memory available.
+LogCompaction with Merge Sort is introduced to achieve lightweight minor compaction by merging records through N-way streaming of ordered data, thus improving the writing performance of the write side. At the same time, thanks to the ordered data, the query performance on the read side can be improved when the primary key is met.
```

Comment:
hi, 
> thus improving the writing performance of the write side

in write side, mean compact action write? not data write? 

> thanks to the ordered data, the query performance on the read side can be improved when the primary key is met

do you mean use `primary key` skipping the logdata which not match, does this mean the statistical information of the main key needs to be preserved?

Reply:
> hi,
> 
> > thus improving the writing performance of the write side
> 
> in write side, mean compact action write? not data write?
> 
> > thanks to the ordered data, the query performance on the read side can be improved when the primary key is met
> 
> do you mean use `primary key` skipping the logdata which not match, does this mean the statistical information of the main key needs to be preserved?

Hello, we don't need to store the statistics of the primary key. The scan range can be pruned using the ordered data